<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FramedReaderSettings</class>
 <widget class="QWidget" name="FramedReaderSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>280</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>3</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QFormLayout" name="formLayout">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::FieldsStayAtSizeHint</enum>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Frame Start:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="CommandEdit" name="leSyncWord">
       <property name="toolTip">
        <string>Enter the 'Frame Start' bytes in hexadecimal.</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string># Channels:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QSpinBox" name="spNumOfChannels">
       <property name="toolTip">
        <string>Select number of channels</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>32</number>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_cm">
       <property name="text">
        <string>Channel Mapping:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QPushButton" name="pbChannelMapping">
       <property name="text">
        <string>Configure Channel Mapping...</string>
       </property>
       <property name="toolTip">
        <string>Configure byte position and data type for each channel</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Total Frame Length:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="spTotalFrameLength">
       <property name="toolTip">
        <string>Enter the total frame length in bytes (including frame start, payload, and checkCode)</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>65535</number>
       </property>
       <property name="value">
        <number>16</number>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_payload">
       <property name="text">
        <string>Payload Size:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QSpinBox" name="spSize">
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="toolTip">
        <string>Calculated automatically: Total Frame Length - Frame Start Length - CheckCode Length</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>65535</number>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>CheckCode:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_checksum">
       <item>
        <widget class="QCheckBox" name="cbChecksum">
         <property name="text">
          <string>Enabled</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pbChecksumConfig">
         <property name="text">
          <string>Configure...</string>
         </property>
         <property name="toolTip">
          <string>Configure checkCode algorithm and byte range</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>1</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="lMessage">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Settings are valid.</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="cbDebugMode">
       <property name="text">
        <string>Debug Mode</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CommandEdit</class>
   <extends>QLineEdit</extends>
   <header>commandedit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
